src/Select.js

```diff
@@ -399,8 +402,14 @@ export default class Select extends Component<Props, State> {
     ) {
       const selectValue = cleanValue(nextProps.value);
       const menuOptions = this.buildMenuOptions(nextProps, selectValue);
-      const focusedValue = this.getNextFocusedValue(selectValue);
-      const focusedOption = this.getNextFocusedOption(menuOptions.focusable);
+      let focusedValue = this.getNextFocusedValue(selectValue);
+      let focusedOption = this.getNextFocusedOption(menuOptions.focusable);
+
+      if (nextProps.autoFocusFirstOption === false  && this.state.focusedOption === null) {
```

Comment:
It is usually better to consult the state directly in the `this.setState()` method as it is asynchronous. Something like 

```js
this.setState((prevState) => ({
  focusedOption: nextProps.autoFocusFirstOption === false  && prevState.focusedOption === null ? null : focusedOption,
  ...
}));
```